### DISCUSSION TITLE
nddev

### DISCUSSION THREAD

**Discussion by ez2rok** [mwermelinger/jollity#4] [Ideas]
Hello. 

I just wanted to make you aware of [nbdev](https://nbdev.fast.ai/), a platform for publishing jupyter notebooks and turning them into packages and blog posts as well. How does you think you package compares to nbdev?

Cheers!

**Comment by mwermelinger** [comment]
Thanks for pointing me to nbdev. 

I see `jollity` as a little tool that can be part of any wider workflow one wishes to have. So, `jollity` would process notebooks (remove spurious blank lines in cells, add headers and footers, number headings, extract Python code, etc.) and the resulting files could then be further processed by GitHub Actions, Quarto, etc. Since `jollity` is a library, one can process the notebooks in different ways at the various stages of the workflow. For example, I do some processing before converting multiple notebooks to a single PDF and some processing after, so that I can hand out both the notebooks and the PDF to my students.
